### PR TITLE
feat(books): implemented User entity and Role entity

### PR DESCRIPTION
**Pull request by YevheniiKilovyi**

### CODE REVIEW

File: src/main/java/com/example/bookstoreapp/model/User.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    )
+    private Set<Role> roles;
+    @Column(name = "is_deleted",
+            nullable = false)
+    private boolean isDeleted = false;
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return roles.stream()
+                .map(r -> new SimpleGrantedAuthority(r.getRoleName().name()))
```
Comment (okuzan) [inline src/main/java/com/example/bookstoreapp/model/User.java]:
```suggestion
                .map(role -> new SimpleGrantedAuthority(role.getRoleName().name()))

```

File: src/main/java/com/example/bookstoreapp/security/jwt/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        System.out.println("getting token by header");
```
Comment (okuzan) [inline src/main/java/com/example/bookstoreapp/security/jwt/JwtAuthenticationFilter.java]: remove

File: src/main/java/com/example/bookstoreapp/security/jwt/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
+            return bearerToken.substring(7);
+        }
```
Comment (okuzan) [inline src/main/java/com/example/bookstoreapp/security/jwt/JwtAuthenticationFilter.java]:
```suggestion
        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith(TOKEN_PREFIX)) {
            return bearerToken.substring(TOKEN_PREFIX.length());
        }

```

File: src/main/java/com/example/bookstoreapp/security/jwt/JwtUtil.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        return Jwts.builder()
+                .setSubject(email)
+                .setIssuedAt(new Date(System.currentTimeMillis()))
+                .setExpiration(new Date(System.currentTimeMillis() + expiration))
+                .signWith(secret)
+                .compact();
+    }
+
+    public boolean isValidToken(String token) {
+        System.out.println("token validation");
```
Comment (okuzan) [inline src/main/java/com/example/bookstoreapp/security/jwt/JwtUtil.java]: use logger instead

File: src/main/java/com/example/bookstoreapp/security/CustomUserDetailsService.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        return userRepository.findUserByEmail(username).orElseThrow(() ->
+                new RegistrationException("Can't find user by username" + username));
```
Comment (okuzan) [inline src/main/java/com/example/bookstoreapp/security/CustomUserDetailsService.java]:
```suggestion
                new EntityNotFoundException("Can't find user by username" + username));
```

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by aaananas — APPROVED** [review]

**Pull request merged**